### PR TITLE
Update arro3 to 0.4

### PR DESCRIPTION
**Pull request by kylebarron**
### Description

Update arro3 from 0.3 to 0.4, ref https://github.com/kylebarron/arro3/releases/tag/py-v0.4.0

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

File: docs/project/changelog.md [hunk 1]
```diff
@@ -84,6 +84,7 @@ myst:
 - Added `rasterio` 1.13.10, `affine` 2.4.0 {pr}`4983`
 - Added `iminuit` 2.29.1 {pr}`4767`, {pr}`5072`
 - Added `arro3-core`, `arro3-io`, and `arro3-compute` 0.3.0 {pr}`5020`
+- Upgraded `arro3-core`, `arro3-io`, and `arro3-compute` 0.4.0 {pr}`5095`
```
Comment (agriyakhetarpal) [inline docs/project/changelog.md:87]:
```suggestion
 - Added `arro3-core`, `arro3-io`, and `arro3-compute` 0.3.0, 0.4.0 {pr}`5020`, {pr}`5095`
```

**Review by agriyakhetarpal — COMMENTED** [review]
Thanks, @kylebarron! You'll need to update the `version:` key for the packages, too, that should fix the CI failure :)

**Comment by kylebarron** [comment]
> You'll need to update the `version:` key for the packages, too, that should fix the CI failure :)

oops, done!

**Review by agriyakhetarpal — APPROVED** [review]

**Comment by agriyakhetarpal** [comment]
No worries, thanks again for the update!

**Pull request merged**